src/Core/3DTiles/C3DTileset.js

```diff
@@ -0,0 +1,113 @@
+import * as THREE from 'three';
+import C3DTBoundingVolume from './C3DTBoundingVolume';
+import C3DTilesTypes from './C3DTilesTypes';
+
+const inverseTileTransform = new THREE.Matrix4();
+
+/** @classdesc
+ * A 3D Tiles
+ *  [Tileset](https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/specification/schema/tileset.schema.json).
+ *  @property {boolean} isC3DTileset - Used by 3D Tiles extensions
+ *  (e.g. {@link C3DTBatchTableHierarchyExtension}) to know in which context
+ *  (i.e. for which 3D Tiles class) the parsing of the extension should be done.
+ *  This value is set to true by default.
+ *  @property {object} asset - Generic information about the tileset, see
+ *  [asset specification]https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/asset.schema.json
+ *  @property {object} properties - Properties associated with the tileset, see
+ *  [tileset specification](https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/tileset.schema.json#L11)
+ *  @property {number} geometricError - see [tileset
+ *  specification](https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/tileset.schema.json#L18)
+ *  @propertu {array} extensionsUsed - see [tileset specification](https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/tileset.schema.json#L27)
+ *  @property {array} extensionsRequired - see [tileset specification](https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/tileset.schema.json#L36)
+ *  @propertiy {object[]} tiles - an array holding all the
+ *  [tiles](https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/specification/schema/tile.schema.json)
+ *  (not their content which is stored in the attribute object3d of the
+ *  layer). This list represents a flattened tileset.
+ *  @property {object} extensions - Extensions of the tileset in the form:
+ * {extensioName1: extensionObject1, extensioName2: extensionObject2, ...}
+ */
+class C3DTileset {
+    constructor(json, baseURL, registeredExtensions) {
+        this.type = C3DTilesTypes.tileset;
```

Comment:
The idea was to use `C3DTilesTypes` when registering an extension to avoid having to write the same string at different places and relating on the fact they are written in the same way to correctly parse extensions; except that I forgot to use it in `3dtiles_batch_table` example... I modified that.

Reply:
`C3DTilesTypes.tileset` it also needs to be written correctly each time and it takes longer to write than `tileset`.

Reply:
Well relying on hard-coded strings instead of an enum is generally a bad idea in my opinion... First, `C3DTilesTypes.tileset` being a variable, you are less likely to make a mistake when writting it (e.g. thanks to you IDE). Then, maintenance is easier in case a value must be changed (you only change it in the enum). Finally, you can also expose to the user what values are available by exporting this enum with associated documentation.

On another note, I can write C3DTilesTypes this way: (which is a bit simpler)

````
const C3DTilesTypes = {
    tileset: {},
    batchtable: {},
    boundingVolume: {},
};
````
Anyway, that is your call, let me know.

Reply:
You're right. Just keep it like that.